.github/workflows/check_code_style.yml

```diff
@@ -13,9 +13,40 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+        with: 
+          submodules: 'true'
 
-      - name: Initialize and update submodule
-        run: git submodule update --init --recursive
+      - name: Check clang-format version
+        id: clang_format_status
+        run: clang-format --version
+        continue-on-error: true
+      
+      - name: Exit if we do not have clang-format installed
+        if: steps.clang_format_status.outcome != 'success'
+        run: |
+          echo "There is no clang-format installed."  
+          exit 1
```

Comment:
Do we need an additional step? If yes, pls add a comment why.

Reply:
You are right, if `clang-format --version` fails => our workflow will fail immediately. So it's redundant , I'll remove it.